torch/csrc/jit/codegen/cuda/type_promotion.cpp

```diff
@@ -195,11 +196,17 @@ std::vector<Val*> promoteValues(
 
 Val* optionalCast(DataType dtype, Val* v) {
   TORCH_INTERNAL_ASSERT(v->getDataType().has_value());
+  // Avoid casting Float and Int scalar values in fusion.
```

Comment:
nitpick, comment doesn't seem to match the implementation. We do "avoid casting Float scalar to specific FloatingPointType", but not "avoid casting float and int scalar values" in general.